google/cloud/bigquerycontrol/integration_tests/job_integration_test.cc

```diff
@@ -0,0 +1,139 @@
+using ::testing::ResultOf;
+
+class BigQueryJobIntegrationTest
+    : public ::google::cloud::testing_util::IntegrationTest {
+ protected:
+  void SetUp() override {
+    project_id_ =
+        google::cloud::internal::GetEnv("GOOGLE_CLOUD_PROJECT").value_or("");
+    ASSERT_FALSE(project_id_.empty());
+  }
+  std::string project_id_;
+};
+
+TEST_F(BigQueryJobIntegrationTest, JobCRUD) {
+  namespace bigquery_proto = google::cloud::bigquery::v2;
+  auto client = JobServiceClient(MakeJobServiceConnectionRest());
+
+  bigquery_proto::JobConfigurationQuery query;
+  query.set_query(
+      "SELECT name, state, year, sum(number) as total "
+      "FROM `bigquery-public-data.usa_names.usa_1910_2013` "
+      "WHERE year >= @minimum_year "
+      "GROUP BY name, state, year");
+
+  // Use GoogleSQL dialect to enable parameterized queries.
+  auto use_legacy_sql = google::protobuf::BoolValue();
+  use_legacy_sql.set_value(false);
+  *query.mutable_use_legacy_sql() = use_legacy_sql;
+  query.set_parameter_mode("NAMED");
+
+  // Specify value for named integer parameter: @minimum_year
+  bigquery_proto::QueryParameter minimum_year_param;
+  auto constexpr kMinimumYearParam = R"pb(
+    name: "minimum_year"
+    parameter_type { type: "INT64" }
+    parameter_value { value { value: "1970" } }
+  )pb";
+  ASSERT_TRUE(google::protobuf::TextFormat::ParseFromString(
+      kMinimumYearParam, &minimum_year_param));
+  *query.add_query_parameters() = minimum_year_param;
+
+  bigquery_proto::JobConfiguration config;
+  *config.mutable_query() = query;
+  bigquery_proto::Job job;
+  *job.mutable_configuration() = config;
+  bigquery_proto::InsertJobRequest job_request;
+  job_request.set_project_id(project_id_);
+  *job_request.mutable_job() = job;
+
+  StatusOr<bigquery_proto::Job> insert_result = client.InsertJob(job_request);
+  ASSERT_STATUS_OK(insert_result);
+  std::string job_id = insert_result->job_reference().job_id();
+  EXPECT_THAT(job_id, Not(IsEmpty()));
+  EXPECT_THAT(
+      insert_result->configuration().query().destination_table().project_id(),
+      Not(IsEmpty()));
+  EXPECT_THAT(
+      insert_result->configuration().query().destination_table().dataset_id(),
+      Not(IsEmpty()));
+  EXPECT_THAT(
+      insert_result->configuration().query().destination_table().table_id(),
+      Not(IsEmpty()));
+
+  google::cloud::bigquery::v2::ListJobsRequest list_request;
+  list_request.set_project_id(project_id_);
+  std::vector<bigquery_proto::ListFormatJob> jobs;
+  for (auto job : client.ListJobs(list_request)) {
+    ASSERT_STATUS_OK(job);
+    jobs.push_back(*std::move(job));
+  }
+  EXPECT_THAT(
+      jobs,
+      Contains(ResultOf(
+          "job id", [](auto const& x) { return x.job_reference().job_id(); },
+          job_id)));
+
+  google::cloud::bigquery::v2::GetJobRequest get_request;
+  get_request.set_project_id(project_id_);
+  get_request.set_job_id(job_id);
+  bool job_complete = false;
+  while (!job_complete) {
```

Comment:
uhhh... can we have a counter that cuts this off after a certain number of attempts?